src/test/java/org/mockito/internal/progress/ArgumentMatcherStorageImplTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.progress;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.internal.matchers.InstanceOf;
+import org.mockitoutil.TestBase;
+
+import static org.junit.Assert.assertEquals;
+
+public class ArgumentMatcherStorageImplTest extends TestBase {
```

Comment:
Instead of a test that tests the internals, please create a regression test in `src/test/java/org/mockitousage` where you copy the example in the issue description of #3331 That way, we test against the user code, rather than how Mockito itself is built.

Reply:
@TimvdLippe 

Thank you for your reply.

Could you please check if I understand correctly?

- Move the test under `src/test/java/org/mockitousage`.
- Modify it so that it does not test against the code inside mockito, but against the user's code (black box testing).